--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -33,7 +33,7 @@ endif()
 set(CMAKE_INSTALL_MESSAGE LAZY)
 
 set(CMAKE_C_STANDARD 11)
-set(CMAKE_CXX_STANDARD 14)
+set(CMAKE_CXX_STANDARD 17)
 set(CMAKE_INSTALL_DEFAULT_COMPONENT_NAME "netdata")
 
 option(USE_CXX_11 "Use C++11 instead of C++14 (should only be used on legacy systems that cannot support C++14, may disable some features)" False)
@@ -191,6 +191,7 @@ option(ENABLE_EXPORTER_MONGODB "Enable e
 option(ENABLE_BUNDLED_JSONC "Force use of a vendored copy of JSON-C" False)
 option(ENABLE_BUNDLED_YAML "Force use of a vendored copy of libyaml" False)
 option(ENABLE_BUNDLED_PROTOBUF "Use a vendored copy of protobuf" False)
+cmake_dependent_option(ENABLE_BUNDLED_ABSEIL "Use a vendored copy of Abseil" False "NOT ENABLE_BUNDLED_PROTOBUF" True)
 
 # Experimental features
 option(ENABLE_WEBRTC "Enable WebRTC dashboard communications (experimental)" False)
--- a/packaging/cmake/Modules/NetdataProtobuf.cmake
+++ b/packaging/cmake/Modules/NetdataProtobuf.cmake
@@ -26,28 +26,12 @@ function(netdata_bundle_protobuf)
         set(FETCHCONTENT_FULLY_DISCONNECTED Off)
 
         if(NEED_ABSL)
-                set(ABSL_PROPAGATE_CXX_STD On)
-                set(ABSL_ENABLE_INSTALL Off)
-                set(BUILD_SHARED_LIBS Off)
-                set(absl_repo https://github.com/abseil/abseil-cpp)
-
-                message(STATUS "Preparing bundled Abseil (required by bundled Protobuf)")
-                if(CMAKE_VERSION VERSION_GREATER_EQUAL 3.28)
-                        FetchContent_Declare(absl
-                                GIT_REPOSITORY ${absl_repo}
-                                GIT_TAG ${ABSL_TAG}
-                                CMAKE_ARGS ${NETDATA_CMAKE_PROPAGATE_TOOLCHAIN_ARGS}
-                                EXCLUDE_FROM_ALL
-                        )
-                else()
-                        FetchContent_Declare(absl
-                                GIT_REPOSITORY ${absl_repo}
-                                GIT_TAG ${ABSL_TAG}
-                                CMAKE_ARGS ${NETDATA_CMAKE_PROPAGATE_TOOLCHAIN_ARGS}
-                        )
+                if(HAVE_ABSEIL AND NOT BUNDLED_ABSEIL)
+                        message(FATAL_ERROR "Using a system copy of Abseil is not supported if vendoring Protobuf")
                 endif()
-                FetchContent_MakeAvailable_NoInstall(absl)
-                message(STATUS "Finished preparing bundled Abseil")
+
+                netdata_bundle_abseil()
+                set(BUNDLED_ABSEIL PARENT_SCOPE)
         endif()
 
         set(protobuf_INSTALL Off)
